mount/mount_linux.go

```diff
@@ -173,6 +174,36 @@ func UnmountAll(mount string, flags int) error {
 	}
 }
 
+// UnmountRecursive unmounts the target and all mounts underneath, starting with
+// the deepest mount first.
+func UnmountRecursive(target string, flags int) (warnings []error, err error) {
```

Comment:
I tried to keep the signature similar to other functions;

- add a `flags` option
- return a `warnings []error` to keep logging out of this package (at least, I _think_ that's the reason this was done in https://github.com/containerd/containerd/blob/ed697290da60cc72390f518d1c9cdde04dd5b078/mount/temp_unix.go#L41-L42


I should note that for the changes in containerd/cri, both are currently unused, so if we want to keep things simple, I can remove the warnings.